From 6bdcd76dae7e45c225794d923619d57efd374dd3 Mon Sep 17 00:00:00 2001
From: XianyuWei <xianyu.wei@nxp.com>
Date: Tue, 24 Jul 2018 14:14:03 +0800
Subject: [PATCH 1/3] Fixed compilation error for the EtherCat drivers.

There were some errors in compiling the module
because EtherCat 1.5.2 did not support the Linux kernel 4.x well.
The 'alloc_netdev' and 'sock_create_kern' interfaces
have been modified in the kernel,
so this patch is added to solve the compilation error.

Signed-off-by: XianyuWei <xianyu.wei@nxp.com>
---
 devices/generic.c | 7 ++++---
 master/ethernet.c | 4 +++-
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/devices/generic.c b/devices/generic.c
index e57bc23..879affb 100644
--- a/devices/generic.c
+++ b/devices/generic.c
@@ -149,7 +149,8 @@ int ec_gen_device_init(
     dev->socket = NULL;
     dev->rx_buf = NULL;

-    dev->netdev = alloc_netdev(sizeof(ec_gen_device_t *), &null, ether_setup);
+	dev->netdev = alloc_netdev(sizeof(ec_gen_device_t *),
+			&null, NET_NAME_UNKNOWN, ether_setup);
     if (!dev->netdev) {
         return -ENOMEM;
     }
@@ -207,8 +208,8 @@ int ec_gen_device_create_socket(
         return -ENOMEM;
     }

-    ret = sock_create_kern(PF_PACKET, SOCK_RAW, htons(ETH_P_ETHERCAT),
-            &dev->socket);
+	ret = sock_create_kern(&init_net, PF_PACKET, SOCK_RAW,
+			htons(ETH_P_ETHERCAT), &dev->socket);
     if (ret) {
         printk(KERN_ERR PFX "Failed to create socket (ret = %i).\n", ret);
         return ret;
diff --git a/master/ethernet.c b/master/ethernet.c
index 2ceea12..7e2032d 100644
--- a/master/ethernet.c
+++ b/master/ethernet.c
@@ -148,7 +148,9 @@ int ec_eoe_init(

     snprintf(eoe->datagram.name, EC_DATAGRAM_NAME_SIZE, name);

-    if (!(eoe->dev = alloc_netdev(sizeof(ec_eoe_t *), name, ether_setup))) {
+	eoe->dev = alloc_netdev(sizeof(ec_eoe_t *), name, NET_NAME_UNKNOWN,
+			ether_setup);
+	if (!eoe->dev) {
         EC_SLAVE_ERR(slave, "Unable to allocate net_device %s"
                 " for EoE handler!\n", name);
         ret = -ENODEV;
--
1.8.3.1

